sdcm/mgmt.py

```diff
@@ -4,11 +4,234 @@
+                                          text=text, task_id=task_id, list_status=list_status, cluster_id=cluster_id, check_task_progress=check_task_progress, timeout=3600)
+        return is_status_reached
+
+    def is_task_status_in_list(self, task_id, cluster_id, list_status, check_task_progress=False):
+        """
+        Check if the status of a given task is in list
+        :param task_id:
+        :param cluster_id:
+        :param list_status:
+        :return:
+        """
+        if check_task_progress:
+            self.get_task_progress(task_id=task_id, cluster_id=cluster_id)
+        return self.get_task_status(task_id=task_id, cluster_id=cluster_id) in list_status
+
+    def wait_for_task_status_done(self, task_id, cluster_id):
+        """
+        Wait and report that task is eventually reaching a 'final' status. meaning one of: done/error/stopped
+        :param task_id:
+        :param cluster_id:
+        :return:
+        """
+        cur_status = self.get_task_status(task_id=task_id, cluster_id=cluster_id)
+        if cur_status == TaskStatus.DONE:
+            return True
+        if cur_status in [TaskStatus.ERROR, TaskStatus.STOPPED]:
+            return False
+
+        if cur_status == TaskStatus.NEW:
+            list_status = [TaskStatus.RUNNING, TaskStatus.ERROR, TaskStatus.DONE]
+            res = self.wait_for_task_status(task_id=task_id, cluster_id=cluster_id, list_status=list_status)
+            if not res:
+                raise ScyllaManagerError("Unexpected result on waiting for task {} status".format(task_id))
+            cur_status = self.get_task_status(task_id=task_id, cluster_id=cluster_id)
+            if cur_status == TaskStatus.ERROR:
+                return False
+
+        if cur_status == TaskStatus.RUNNING:
+            list_status = [TaskStatus.DONE, TaskStatus.ERROR]
+            res = self.wait_for_task_status(task_id=task_id, cluster_id=cluster_id, list_status=list_status, check_task_progress=True)
+            if not res:
+                raise ScyllaManagerError("Unexpected result on waiting for task {} status {}".format(task_id, list_status))
+            cur_status = self.get_task_status(task_id=task_id, cluster_id=cluster_id)
+            if cur_status == TaskStatus.ERROR:
+                return False
+
+        return cur_status == TaskStatus.DONE
+
+    def _mgr_remoter_run(self, cmd):
+        result = self.manager_node.remoter.run(cmd)
+        logger.debug("[_mgr_remoter_run]: result: {} , result.stdout: {} , result.stderr: {}".format(result,result.stdout, result.stderr))
+        return result
+
+    def _run_sctool_cmd(self, cmd):
+        logger.debug("Issuing sctool command: {}".format(cmd))
+        return self._mgr_remoter_run(cmd='sudo sctool {}'.format(cmd))
+
+    def get_cluster(self, cluster_name):
+        resp = self._run_sctool_cmd('cluster list')
+        cluster_id = "TODO" #TODO
```

Comment:
I don't understand what is the purpose of `cluster_id`

Reply:
done